types/errors.go

```diff
@@ -0,0 +1,178 @@
+var ErrOldReport = errors.New("More recent report already exists in storage")
+
+// ItemNotFoundError shows that item with id ItemID wasn't found in the storage
+type ItemNotFoundError struct {
+	ItemID interface{}
+}
+
+// Error returns error string
+func (e *ItemNotFoundError) Error() string {
+	return fmt.Sprintf("Item with ID %+v was not found in the storage", e.ItemID)
+}
+
+// TableNotFoundError table not found error
+type TableNotFoundError struct {
+	tableName string
+}
+
+// Error returns error string
+func (err *TableNotFoundError) Error() string {
+	return fmt.Sprintf("no such table: %v", err.tableName)
+}
+
+// TableAlreadyExistsError represents table already exists error
+type TableAlreadyExistsError struct {
+	tableName string
+}
+
+// Error returns error string
+func (err *TableAlreadyExistsError) Error() string {
+	return fmt.Sprintf("table %v already exists", err.tableName)
+}
+
+// ForeignKeyError something violates foreign key error
+// tableName and foreignKeyName can be empty for DBs not supporting it (SQLite)
+type ForeignKeyError struct {
+	TableName      string
+	ForeignKeyName string
+
+	// Details can reveal you information about specific item violating fk
+	Details string
+}
+
+// Error returns error string
+func (err *ForeignKeyError) Error() string {
+	return fmt.Sprintf(
+		`operation violates foreign key "%v" on table "%v"`, err.ForeignKeyName, err.TableName,
+	)
+}
+
+// ConvertDBError converts sql errors to those defined in this package
+func ConvertDBError(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	err = convertPostgresError(err)
+	err = convertSQLiteError(err)
+
+	return err
+}
+func regexGetFirstMatchOrLogError(regexStr string, str string) string {
+	return regexGetNthMatchOrLogError(regexStr, 1, str)
+}
+
+func regexGetNthMatchOrLogError(regexStr string, nMatch uint, str string) string {
+	match, err := regexGetNthMatch(regexStr, nMatch, str)
+	if err != nil {
+		log.Error().
+			Str("regex", regexStr).
+			Str("str", str).
+			Msgf("unable to get first match from string '%v' with regex '%v'", str, regexStr)
+		return ""
+	}
+
+	return match
+}
+func regexGetFirstMatch(regexStr string, str string) (string, error) {
+	return regexGetNthMatch(regexStr, 1, str)
+}
+
+func regexGetNthMatch(regexStr string, nMatch uint, str string) (string, error) {
+	regex := regexp.MustCompile(regexStr)
+	if !regex.MatchString(str) {
+		return "", errors.New("regex doesn't match string")
+	}
+
+	matches := regex.FindStringSubmatch(str)
+	if len(matches) < int(nMatch+1) {
+		return "", errors.New("regexGetNthMatch unable to find match")
+	}
+
+	return matches[nMatch], nil
+}
+
+func convertPostgresError(err error) error {
+	pqError, ok := err.(*pq.Error)
+	if !ok {
+		return err
+	}
+
+	// see https://www.postgresql.org/docs/current/errcodes-appendix.html to get the magic happening below
+	switch pqError.Code {
+	case "42P07": // duplicate_table
```

Comment:
prob. worth to use `const` there in the future. But let's not delay this fix to be deployed.